### DISCUSSION TITLE
[Вопрос] Стандартные теги для девичьих фамилий и прочего

### DISCUSSION THREAD

**Discussion by rilian-la-te** [Serg-Norseman/GEDKeeper#548] [Q&A]
Какие стандартные теги существуют в GED для девичьих фамилий, отчеств и других славянских генеалогических изысков? И может ли их поддерживать программа? Интересует в основном общение с gramps и Genotek.

**Comment by Serg-Norseman** [comment]
Стандартных тегов для всего этого нет совсем никаких. Точно (возможно в GEDCOM 7.X что-то изменилось, последний раз туда заглядывал с год назад).

Есть множество тегов, которые вводились в различных других программах. Я в основном своего не выдумывал, встраивал поддержку таких подходящих тегов, которые находил в других программах. 

Например, для фамилий в замужестве _MARNM & _MARN (девичью другие программы чаще пишут в стандартный тег (вероятно потому, что у множества народов женщина не меняет фамилию в браке), я сделал также, хотя и другие варианты бывают). Для отчеств есть тег _PATN, хотя по прошествии времени, мне кажется что лучше было сделать некий усредненный вариант и использовать _MIDN (приравнять внутри программы отчество к среднему имени в части европейских стран).

Фамилии в замужестве поддерживаются программой при включении опций расширенных женских фамилий. Там же можно выбрать способ отображения обоих фамилий - что в каком порядке. Опция была сделана для вариативности, т.к. не все хотят использовать такое. Я, к примеру, пишу девичью и в замужестве в одно поле фиксированно, в формате "Девичья (вЗамужестве1, вЗамужестве2, ...)", т.е. без опций - мне так удобно и достаточно. А кому-то нравится в разбивку.

По Gramps и Genotek - совместимость проверяется только по конкретным запросам. Специально в постоянном режиме не проверяется, т.к. запросы бывают редко, а времени и усилий на такое нужно много. При наличии участника проекта, кто занимался бы этим на постоянной основе - было бы проще.

**Reply by hazzik** [reply]
> Для отчеств есть тег _PATN, хотя по прошествии времени, мне кажется что лучше было сделать некий усредненный вариант и использовать _MIDN (приравнять внутри программы отчество к среднему имени в части европейских стран).

Лучше, так, наверное, не делать.

**Reply by Serg-Norseman** [reply]
Да, момент очень спорный. Иногда мне кажется то, что я писал выше, чаще - что всё сделано приемлемо и ничего менять не нужно... Думаю забугорных пользователей из стран, где отчества нет - эта надпись изрядно смущает. Хотя никто не мешает в переводах для таких стран вместо "отчество" написать "среднее имя". По смыслу и типовому функционалу GEDCOM будет одинаково.

**Reply by hazzik** [reply]
У меня два сына, у одного есть среднее имя, у другого отчество. У обоих есть русскоязычные документы.

**Reply by Serg-Norseman** [reply]
Еще была мысль для таких случаев просто расширить функционал суррогатных "культур", которые введены в GEDKeeper - там уже есть признаки, есть ли в культуре отчество и фамилия. И частично это отрабатывается на уровне интерфейса. Так что можно расширить и сделать переключение на "среднее имя", где оно есть.

**Reply by rilian-la-te** [reply]
Кстати, а как выбирать культуру в GEDKeeper?

**Reply by Serg-Norseman** [reply]
Культуры привязаны к языку имён и определяют наличие отчества, фамилии, системную кодировку для нужд сортировки и сравнения. Список добавленных культур небольшой и весь присутствует в папке locales\cultures.

Чтобы конкретная культура действовала на функционал программы - нужно выбрать необходимый язык для всей БД в свойствах файла или в любом конкретном имени.

**Comment by hazzik** [comment]
> Какие стандартные теги существуют в GED для девичьих фамилий, отчеств и других славянских генеалогических изысков?

Если мы говорим, про стандард, то он предписывает на каждую смену имени иметь отдельный тег `NAME`. Внутри может быть структура с указанием частей имени, типа, переводов (в 7 версии), и т.д. Стандартного тега для среднего имени и отчества нет.

```gedcom
1 NAME Мария Ивановна /Петрова/
2 TYPE BIRTH # or MAIDEN
# 2 NPFX
2 GIVN Мария
# 2 NICK
# 2 SPFX
2 SURN Петрова
# 2 NSFX

1 NAME Мария Ивановна /Кузнецова/
2 TYPE MARRIED
2 GIVN Мария
2 SURN Кузнецова
```

> И может ли их поддерживать программа?

Программа работает с отчетсвами, через под-тег `_PATN`

**Reply by rilian-la-te** [reply]
Генотек не кушает девичьи фамилии из GEDKeeper, а GRAMPS не ест ни девичьих фамилий, ни отчеств. Потому и вопрос возник. Проверялась последняя версия GEDKeeper из git, и последний gramps из репозитория ArchLinux.

**Reply by Serg-Norseman** [reply]
Надо или искать информацию по формату файлов в грампс и генотек, или делать родные для них файлы, вводить туда нужную специфическую информацию, прогонять импорт/экспорт в них, чтобы увидеть корректную загрузку родного для них формата, и вот после этого - сравнивать их формат файлов с форматом GEDKeeper, где что не совпадает.

**Reply by rilian-la-te** [reply]
genotek не умеет выгружать в gedcom (только загружать), а с gramps можно попробовать. Gramps опенсорсный, так что можно просто прочитать, как там работает парсер GEDCom. Вот сорцы ихнего парсера (если Питон знаете, то будет понятно https://github.com/gramps-project/gramps/blob/master/gramps/plugins/lib/libgedcom.py)

**Reply by rilian-la-te** [reply]
Например, gramps держит девичью фамилию как основную, а вторичную предлагает хранить в теге __MARN (https://github.com/gramps-project/gramps/blob/master/gramps/plugins/lib/libgedcom.py#L4807)

**Reply by Serg-Norseman** [reply]
У них _MARNM, в GEDKeeper - _MARN на сохранение, и оба _MARNM+_MARN на чтение. Чтобы подстроиться под Gramps, придется делать специальный режим экспорта с тем тегом, который он поддерживает. На это нужен ресурс, сейчас ресурса нет, загрузка большая. Так что если в перспективе.

**Reply by Serg-Norseman** [reply]
Просмотрел внимательнее исходник GRAMPS - странно, вообще я вижу, что они должны понимать все три тега для фамилий в замужестве: _MAR, _MARN, _MARNM.

**Reply by rilian-la-te** [reply]
Я еще раз попробую, возможно, у меня старая версия была. А вот с отчествами там засада, разве что им issue создавать.
